scripts/zen_workaround.py

```diff
@@ -53,11 +53,26 @@ def is_secure_boot_enabled():
         sys.exit(ret)
 
 def read_msr(cpu):
-    msr = os.open('/dev/cpu/{}/msr'.format(cpu), os.O_RDONLY)
-    os.lseek(msr, MSR, os.SEEK_SET)
-    (val,) = struct.unpack('<q', os.read(msr, 8))
-    os.close(msr)
-    return val
+    try:
+        msr = os.open('/dev/cpu/{}/msr'.format(cpu), os.O_RDONLY)
+    except PermissionError:
+        sys.stderr.write(str(e) + '\n')
+        print("Permission denied opening MSR for reading.")
+        print("Try running as root / with sudo.")
+        return 0
+    try:
+        os.lseek(msr, MSR, os.SEEK_SET)
+        (val,) = struct.unpack('<q', os.read(msr, 8))
+        return val
+    except PermissionError as e:
+        sys.stderr.write(str(e) + '\n')
+        print("Permission denied on reading from MSR.")
+    except OSError as e:
+        sys.stderr.write(str(e) + '\n')
+        print("General error on reading from MSR.")
+        return 0
```

Comment:
Why is returning 0 a good idea? Shouldn't we kill the script?

Reply:
Returning 0 runs into "not supported" later on and is similar what was done in the other place.
If you want to I can drop the return 0 and break / continue which we have into killing the script after the error (which also may be placed into stderr completely).